src/test/scala/nl.knaw.dans.easy.deposit/EasyDepositApiAppSpec.scala

```diff
@@ -15,27 +15,93 @@
  */
 package nl.knaw.dans.easy.deposit
 
+import java.nio.file.attribute.PosixFilePermission
 import java.util.UUID
 
-import nl.knaw.dans.easy.deposit.docs.DepositInfo
+import nl.knaw.dans.easy.deposit.Errors.NoSuchDepositException
 import nl.knaw.dans.easy.deposit.docs.StateInfo.State
 import nl.knaw.dans.easy.deposit.docs.StateInfo.State.State
-import org.apache.commons.configuration.PropertiesConfiguration
+import nl.knaw.dans.easy.deposit.docs.{ DepositInfo, StateInfo }
+import nl.knaw.dans.lib.error._
+import org.apache.commons.configuration.{ ConfigurationException, PropertiesConfiguration }
 import org.joda.time.DateTime
 import org.joda.time.DateTimeZone.UTC
 
-import scala.util.Success
-
+import scala.util.{ Failure, Success }
 class EasyDepositApiAppSpec extends TestSupportFixture {
   private val app: EasyDepositApiApp = createTestApp()
 
-  private val defaultUser: String = "user001"
-
   override def beforeEach(): Unit = {
     super.beforeEach()
     clearTestDir()
   }
 
+  private val oldState: StateInfo = StateInfo(State.draft, "Deposit is open for changes.")
+  private val newState = StateInfo(State.archived, "blabla")
+  private val expectedStates = {
+    Success(
+      """OLD: {"state":"DRAFT","stateDescription":"Deposit is open for changes."}
+        |NEW: {"state":"ARCHIVED","stateDescription":"blabla"}
+        |""".stripMargin
+    )
+  }
+
+  "forceChangeState" should "report a missing deposit" in {
+    app.forceChangeState(
+      defaultUserInfo.id,
+      uuid,
+      StateInfo(State.submitted, "blabla"),
+      update = false,
+    ) should matchPattern {
+      case Failure(e: NoSuchDepositException) if e.getMessage == s"Deposit $uuid not found" =>
+    }
+  }
+
+  it should "show the old and new state" in {
+    val uuid = app.createDeposit(defaultUserInfo.id).getOrRecover(e => fail(e)).id // preparation
+
+    app.forceChangeState(defaultUserInfo.id, uuid, newState, update = false) shouldBe expectedStates
+    app.getDepositState(defaultUserInfo.id, uuid) shouldBe Success(oldState) // post condition
+  }
+
+  it should "alter the state" in {
+    val uuid = app.createDeposit(defaultUserInfo.id).getOrRecover(e => fail(e)).id // preparation
+
+    app.forceChangeState(defaultUserInfo.id, uuid, newState, update = true) shouldBe expectedStates
+    app.getDepositState(defaultUserInfo.id, uuid) shouldBe Success(newState) // post condition
+  }
+
+  it should "cause trouble when setting the state from draft to submitted" in {
```

Comment:
It is even more dangerous when the deposit has been submitted before. Then there will be `bag-store.bag-id` but that is related to the previous attempt.